### DISCUSSION TITLE
Setting environment variable

### DISCUSSION THREAD

**Discussion by Abhi-SimuTech** [ansys/pyaedt#3741] [Q&A] [answered]
Hi,

I have been trying to create a geometry of an array with a very large number of elements using PyAEDT scripting.
The issue is, The when the number of elements are few thousands, the geometry creation (through script) fails sometimes; but when it is just a few 10s of elements, it works perfectly at all times. (Btw, all the elements are identical.)

In the Message Manager, HFSS suggests to set the environment variable in a certain way as an work around. (Please see the message below.) Wondering what would be the PyAEDT function to implement this!

![image](https://github.com/ansys/pyaedt/assets/145813888/68eed6f1-fc1e-4b98-b449-13b3f886bf84)

Regards
Abhi

**Comment by MaxJPRey** [comment, marked as answer]
Hi @Abhi-SimuTech,
You would just have to do it as you do in Python for any other application

```
import os
os.environ["ANS_USE_ISOLATED_CLIPBOARD"] = "1"
```

Note: Please be cautious in doing so and keep in mind that it can change the behavior of following simulations.

The global environment variable outside of the PyAEDT application will not affected. If you want to persist an environment variable across different processes or system-wide, you need to set it in the operating system settings or in a script that is executed when the system starts or when a new shell is started.
